### PR TITLE
[DOCU-2308] DecK Konnect guide

### PR DESCRIPTION
**Pull request by lena-larionova**
### Summary

Adding a guide for using decK with Konnect with examples into the decK docs.

The only thing I couldn't figure out was the purpose of the `--konnect-addr` flag in the current environment. Users only have one API they can target, right? Or can they somehow target `konnect.konghq.com` still? I couldn't get it to do anything different - @GGabriele, is there something I'm missing, or is this flag not helpful to customers at the moment (only useful internally)?

Note that this is going into live docs, not into the `konnect` branch, because this version of decK is already out - even though Konnect is in beta.

### Reason
https://konghq.atlassian.net/browse/DOCU-2308

### Testing
Tested all the things with decK 1.12 and cloud.konghq.com; that's also how I figured out precedence for auth methods. If that's not the desired precedence, then it's something that we should know about anyway. 

Preview: https://deploy-preview-4030--kongdocs.netlify.app/deck/latest/guides/konnect/

### CODE REVIEW

**Comment by GGabriele** [comment]
> The only thing I couldn't figure out was the purpose of the --konnect-addr flag in the current environment. Users only have one API they can target, right? Or can they somehow target konnect.konghq.com still? I couldn't get it to do anything different 

Two reasons:
- target the dev environment (internal only)
- target konnect.konghq.com

File: src/deck/guides/konnect.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+decK automatically uses the credentials in `~/.deck.yaml` in any subsequent
+calls:
+
+```
+deck ping
+
+Successfully Konnected as MyName (Konnect Org)!
+```
+
+## Runtime groups
```
Comment (GGabriele) [inline src/deck/guides/konnect.md:113]: May be worth adding an example using `--konnect-runtime-group-name` to dump a config from a non-default runtime group?

**Comment by lena-larionova** [comment]
> Two reasons:
target the dev environment (internal only)
target konnect.konghq.com

We won't be documenting how to target the internal one, so that's fine. 
To target konnect.konghq.com, is it simply `--konnect-addr konnect.konghq.com`?

**Comment by GGabriele** [comment]
> > Two reasons:
> > target the dev environment (internal only)
> > target konnect.konghq.com
> 
> We won't be documenting how to target the internal one, so that's fine. To target konnect.konghq.com, is it simply `--konnect-addr konnect.konghq.com`?

`--konnect-addr https://konnect.konghq.com`

File: src/deck/guides/konnect.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+1. Password set with the `--konnect-password` flag
+2. decK configuration file (default: `~/.deck.yaml`)
+3. File passed with the `--konnect-password-file` flag
+
+For example, if you have both a decK config file and a {{site.konnect_short_name}} password file, decK uses the password in the config file.
+
+### Authenticate with CLI flags
+
+There are two ways to authenticate through CLI flags: by passing the password in `--konnect-password` directly, or by saving it to a file and passing the filename to decK with `--konnect-password-file`.
```
Comment (cloudjumpercat) [inline src/deck/guides/konnect.md:53]:
```suggestion
There are two ways to authenticate through CLI flags. You can pass the password in `--konnect-password` directly or save it to a file and pass the filename to decK with `--konnect-password-file`.
```
Just reworked this a bit to see if I could remove the comma between the dependent clauses (I think that's right, but not 100% sure).

File: src/deck/guides/konnect.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+With `konnect-password`:
```
Comment (cloudjumpercat) [inline src/deck/guides/konnect.md:55]:
```suggestion
The following example shows how to use `konnect-password` to authenticate:
```

File: src/deck/guides/konnect.md [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+```sh
+deck ping \
+  --konnect-email YOUR_EMAIL \
+  --konnect-password YOUR_PASSWORD
+```
+
+Or, save your {{site.konnect_short_name}} password to a file, then pass the file using the `--konnect-password-file` flag:
```
Comment (cloudjumpercat) [inline src/deck/guides/konnect.md:63]:
```suggestion
The following example shows how to save your {{site.konnect_short_name}} password to a file, then pass the file using the `--konnect-password-file` flag:
```

File: src/deck/guides/konnect.md [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+
+### Authenticate with CLI flags
+
+There are two ways to authenticate through CLI flags: by passing the password in `--konnect-password` directly, or by saving it to a file and passing the filename to decK with `--konnect-password-file`.
+
+With `konnect-password`:
+
+```sh
+deck ping \
+  --konnect-email YOUR_EMAIL \
```
Comment (cloudjumpercat) [inline src/deck/guides/konnect.md:59]: Should this be "email@example.com" based on our discussion about placeholder examples?

File: src/deck/guides/konnect.md [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+  --konnect-password YOUR_PASSWORD
+```
+
+Or, save your {{site.konnect_short_name}} password to a file, then pass the file using the `--konnect-password-file` flag:
+
+```sh
+deck ping \
+  --konnect-email YOUR_EMAIL \
```
Comment (cloudjumpercat) [inline src/deck/guides/konnect.md:67]: Same comment here about if this should be "email@example.com" or something like that.

File: src/deck/guides/konnect.md [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+The default decK configuration file is `~/.deck.yaml`.
+You can store {{site.konnect_short_name}} credentials in this file to pass them more securely.
+For example:
+
+```
+konnect-email: example@email.com
+konnect-password: YOUR_PASSWORD
+```
+
+Alternatively, you can save your password in a separate file, then specify the password file instead of a password:
```
Comment (cloudjumpercat) [inline src/deck/guides/konnect.md:82]:
```suggestion
Alternatively, you can save your password in a separate file, then specify the password file instead of a password. 
For example:
```
Just to mirror the one above.

File: src/deck/guides/konnect.md [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+
+```
+konnect-email: example@example.com
+konnect-password-file: PATH/TO/FILENAME
+```
+
+decK automatically uses the credentials in `~/.deck.yaml` in any subsequent calls:
```
Comment (cloudjumpercat) [inline src/deck/guides/konnect.md:89]:
```suggestion
decK automatically uses the credentials in `~/.deck.yaml` in any subsequent calls.
For example:
```

File: src/deck/guides/konnect.md [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+
+If your account is in the `konnect.konghq.com` environment, use this flag to target the relevant API:
+
+```sh
+deck ping --konnect-addr https://konnect.konghq.com
+```
+
+## Runtime groups
+
+Target runtime groups in your state file with the `konnect_runtime_group` parameter:
```
Comment (cloudjumpercat) [inline src/deck/guides/konnect.md:112]: If this is an option you can add to your basic config file, I might add this as a bullet under step one with something like "If you want to target runtime groups....". I just wasn't sure how this related to the steps below.

File: src/deck/guides/konnect.md [hunk 10]
```diff
@@ -0,0 +1,227 @@
...
+
+```yaml
+_format_version: "1.1"
+_konnect:
+  runtime_group_name: staging
+```
+
+You can also set this parameter using the `--konnect-runtime-group-name` flag:
```
Comment (cloudjumpercat) [inline src/deck/guides/konnect.md:120]: Same comment as the one above about if this could be added as an optional "you could also add this" bullet under step one.

File: src/deck/guides/konnect.md [hunk 11]
```diff
@@ -0,0 +1,227 @@
...
+```sh
+deck sync --konnect-runtime-group-name default
+```
+
+A state file can only target one runtime group.
+
+If you leave this empty, don't provide a flag, or don't include a `_konnect` section at all, decK targets the `default` runtime group.
+You can see this by pushing a config file to {{site.konnect_short_name}} with `deck sync`.
+
+1. Create a basic config file:
```
Comment (cloudjumpercat) [inline src/deck/guides/konnect.md:131]:
```suggestion
1. Create a basic config file. For example:
```
Just to match other task lists I've seen where the command ends in a period. Not sure if that's standard or not.

File: src/deck/guides/konnect.md [hunk 12]
```diff
@@ -0,0 +1,227 @@
...
+
+    ```yaml
+    _format_version: "1.1"
+    services:
+    - name: example_service
+      host: mockbin.org
+    ```
+
+2. Sync the file to {{site.konnect_short_name}}:
```
Comment (cloudjumpercat) [inline src/deck/guides/konnect.md:140]:
```suggestion
2. Sync the file to {{site.konnect_short_name}}.
```

File: src/deck/guides/konnect.md [hunk 13]
```diff
@@ -0,0 +1,227 @@
...
+
+    ```sh
+    deck sync
+    ```
+
+3. Pull down the configuration with `deck dump`:
```
Comment (cloudjumpercat) [inline src/deck/guides/konnect.md:146]:
```suggestion
3. Pull down the configuration with `deck dump`.
```

File: src/deck/guides/konnect.md [hunk 14]
```diff
@@ -0,0 +1,227 @@
...
+
+    ```sh
+    deck dump --konnect-runtime-group-name default
+    ```
+
+    See the newly added `_konnect` section:
```
Comment (cloudjumpercat) [inline src/deck/guides/konnect.md:152]:
```suggestion
    See the newly added `_konnect` section. For example:
```
One question I had is why they need to see the new section. Do they need to use that info in another spot?

File: src/deck/guides/konnect.md [hunk 15]
```diff
@@ -0,0 +1,227 @@
...
+
+## {{site.konnect_short_name}} service tags
+
+In {{site.konnect_short_name}}, there are two types of services:
+* Gateway services: Managed through Runtime Manager
+* {{site.konnect_short_name}} services: Managed through Service Hub
+
+decK manages Gateway services, which contain configurations for the Gateway proxy.
+
+Although decK doesn't _directly_ manage {{site.konnect_short_name}} services, you can use tags to associate a Gateway service to a {{site.konnect_short_name}} service:
```
Comment (cloudjumpercat) [inline src/deck/guides/konnect.md:171]:
```suggestion
Although decK doesn't _directly_ manage {{site.konnect_short_name}} services, you can use tags to associate a Gateway service to a {{site.konnect_short_name}} service. For example:
```

File: src/deck/guides/konnect.md [hunk 16]
```diff
@@ -0,0 +1,227 @@
...
+### Authentication with a {{site.konnect_short_name}} password file is not working
+
+If you have verified that your password is correct but decK can't connect to your account, check for conflicts with the decK config file (`~/.deck.yaml`) and the {{site.konnect_short_name}} password file.
+There is likely a decK config file conflicting with the password file and passing another set of credentials.
+
+To resolve, remove one of the duplicate sets of credentials.
+
+### Workspace connection refused
+
+When migrating from {{site.base_gateway}} to {{site.konnect_short_name}}, make sure to remove any `_workspace`tags. If you leave `_workspace` in, you get the following error:
```
Comment (cloudjumpercat) [inline src/deck/guides/konnect.md:205]:
```suggestion
When migrating from {{site.base_gateway}} to {{site.konnect_short_name}}, make sure to remove any `_workspace` tags. If you leave `_workspace` in, you get the following error:
```

File: src/deck/guides/konnect.md [hunk 17]
```diff
@@ -0,0 +1,227 @@
...
+
+```
+Error: checking if workspace exists
+```
+
+Remove the `_workspace` key to resolve this error.
+
+You can now sync the file as-is to apply it to the default runtime group, or add a key to apply the configuration to a specific runtime group.
+
+To apply configuration to custom runtime groups, replace `_workspace` with `runtime_group_name: GroupName`.
```
Comment (cloudjumpercat) [inline src/deck/guides/konnect.md:215]:
```suggestion
To apply the configuration to custom runtime groups, replace `_workspace` with `runtime_group_name: GroupName`.
```

File: src/deck/guides/konnect.md [hunk 18]
```diff
@@ -0,0 +1,227 @@
...
+
+For example, to export the configuration from workspace `staging` to runtime group `staging`, you would change:
+
+```
+_workspace: staging
+```
+
+Into:
```
Comment (cloudjumpercat) [inline src/deck/guides/konnect.md:223]:
```suggestion
To:
```
Could be a personal preference though, so feel free to ignore.

File: src/deck/guides/konnect.md (hunk 3 above)
Comment (lena-larionova) [reply src/deck/guides/konnect.md:55]: This is a bit wordy, but I can see where you're going with it. I'll try to reword to add a bit more context to this and the next example.

File: src/deck/guides/konnect.md (hunk 9 above)
Comment (lena-larionova) [reply src/deck/guides/konnect.md:112]: It's more about targeting custom runtime groups. By default, a file will target the runtime group `default`, and if you want to change that, you have to set it either in the file or with a flag. Even if you don't explicitly set it though, decK will add the parameter.

File: src/deck/guides/konnect.md (hunk 14 above)
Comment (lena-larionova) [reply src/deck/guides/konnect.md:152]: In hindsight, this entire section is irrelevant. It's literally just here to illustrate what happens if you don't provide a group, but there's no reason to document that with steps.

**Comment by lena-larionova** [comment]
@cloudjumpercat Reworked a bunch of things, your feedback made me realize that some of the content was organized in a confusing way. 

I removed the steps with the `deck sync` and `deck dump` examples, as they weren't really adding any value - the point of this topic is to serve as a reference, and there's no reason to do those steps other than to validate decK's behavior. I also split out the two CLI flag authentication options; not really sure why I had them lumped together in the first place.

File: src/deck/guides/konnect.md (hunk 11 above)
Comment (lena-larionova) [reply src/deck/guides/konnect.md]: Ack - that's just inconsistency on my part.

File: src/deck/guides/konnect.md [hunk 19]
```diff
@@ -0,0 +1,199 @@
...
+You can also [save {{site.konnect_short_name}} credentials to a decK config file](#authenticate-with-a-deck-config-file) to target {{site.konnect_short_name}} by default.
+
+If you don't pass any {{site.konnect_short_name}} parameters to decK, decK looks for a local {{site.base_gateway}} instance instead.
+
+{:.note}
+> Prior to decK 1.12, decK provided `deck konnect` commands.
+These commands are deprecated and have been replaced with flags.
+
+`--konnect-addr`
+:  Address of the {{site.konnect_short_name}} endpoint. (Default: `"https://us.api.konghq.com"`)
```
[the remark this answers is not in the document]
Comment (lena-larionova) [reply src/deck/guides/konnect.md:26]: ^ That's what I'm going with for now. Keeping the doc as-is till we fix it in the source.

File: src/deck/guides/konnect.md [hunk 20]
```diff
@@ -0,0 +1,199 @@
...
+
+`--konnect-runtime-group-name`
+:  {{site.konnect_short_name}} runtime group name.
+
+## Authenticate with {{site.konnect_short_name}}
+
+decK looks for {{site.konnect_short_name}} credentials in the following order of precedence:
+
+1. Password set with the `--konnect-password` flag
+2. decK configuration file (default: `~/.deck.yaml`)
```
[the remark this answers is not in the document]
Comment (lena-larionova) [reply src/deck/guides/konnect.md:46]: The latter -  decK looks for a .deck.yaml file in the home directory, it doesn't create one. Do you think it would be useful to add that clarification?

File: src/deck/guides/konnect.md [hunk 21]
```diff
@@ -0,0 +1,199 @@
...
+
+### Authenticate using a password file
+
+For a more secure approach, you can save your {{site.konnect_short_name}}
+password to a file, then pass the filename to decK with `--konnect-password-file`:
+
+```sh
+deck ping \
+  --konnect-email example@example.com \
+  --konnect-password-file /PATH/TO/FILE
```
[the remark this answers is not in the document]
Comment (lena-larionova) [reply src/deck/guides/konnect.md:69]: It doesn't! I tested with a couple of different things (no extension, .txt, .yaml, .md) and they all worked.

File: src/deck/guides/konnect.md [hunk 22]
```diff
@@ -0,0 +1,201 @@
...
+* Gateway services: Managed through Runtime Manager
+* {{site.konnect_short_name}} services: Managed through Service Hub
+
+decK manages Gateway services, which contain configurations for the Gateway proxy.
+
+Although decK doesn't _directly_ manage {{site.konnect_short_name}} services, you can use tags to associate a Gateway service to a {{site.konnect_short_name}} service:
+
+```yaml
+tags:
+- _KonnectService:example
```
Comment (haydenlam) [inline src/deck/guides/konnect.md:149]:
```suggestion
- _KonnectService:<Konnect-Service-Name>
```

File: src/deck/guides/konnect.md [hunk 23]
```diff
@@ -0,0 +1,201 @@
...
+```
+If the {{site.konnect_short_name}} service doesn't exist, setting this tag creates a {{site.konnect_short_name}} service.
```
Comment (haydenlam) [inline src/deck/guides/konnect.md:151]:
```suggestion
The `<Konnect-Service-Name>` identifies which {{site.konnect_short_name}} to associate the Gateway service to. If the {{site.konnect_short_name}} service doesn't exist, setting this tag creates a {{site.konnect_short_name}} service.
```

File: src/deck/guides/konnect.md [hunk 24]
```diff
@@ -0,0 +1,201 @@
...
+
+    ```sh
+    deck sync --konnect-runtime-group-name staging
+    ```
+
+## {{site.konnect_short_name}} service tags
+
+In {{site.konnect_short_name}}, there are two types of services:
+* Gateway services: Managed through Runtime Manager
+* {{site.konnect_short_name}} services: Managed through Service Hub
```
Comment (haydenlam) [inline src/deck/guides/konnect.md:141]:
```suggestion
* {{site.konnect_short_name}} services: Managed through Service Hub
Each Konnect Service may contain one or more service versions. A service version represents an implementation of a Gateway service.
```

File: src/deck/guides/konnect.md [hunk 25]
```diff
@@ -0,0 +1,201 @@
...
+
+decK manages Gateway services, which contain configurations for the Gateway proxy.
+
+Although decK doesn't _directly_ manage {{site.konnect_short_name}} services, you can use tags to associate a Gateway service to a {{site.konnect_short_name}} service:
```
Comment (haydenlam) [inline src/deck/guides/konnect.md:145]:
```suggestion
Although decK doesn't _directly_ manage {{site.konnect_short_name}} services or service versions, you can use tags to associate a Gateway service to a service version in a {{site.konnect_short_name}} service:
```
